crates/optimism/txpool/src/transaction.rs

```diff
@@ -287,6 +294,14 @@ where
 pub trait OpPooledTx:
     MaybeConditionalTransaction + MaybeInteropTransaction + PoolTransaction + DataAvailabilitySized
 {
+    /// Returns the EIP-2718 encoded bytes of the transaction.
+    fn encoded_2718(&self) -> Cow<'_, Bytes> {
+        let mut encoded = Vec::with_capacity(self.encoded_length());
+        let tx = self.clone_into_consensus();
+        tx.encode_2718(&mut encoded);
+
+        Cow::Owned(Bytes::copy_from_slice(&encoded))
+    }
 }
 impl<T> OpPooledTx for T where
```

Comment:
this means that we can no longer auto impl this and instead need to manually impl this for OpPooledTransaction